diff --git a/lib/typescript/src/types.d.ts b/lib/typescript/src/types.d.ts
index a8d791964ddd6d0a92127e0fa7e59b27fed232a5..1220f7de6981f28fa2edf701d8454f56eb6d192a 100644
--- a/lib/typescript/src/types.d.ts
+++ b/lib/typescript/src/types.d.ts
@@ -82,7 +82,7 @@ export type NavigationAction = Readonly<{
     /**
      * Additional data for the action
      */
-    payload?: object;
+    payload?: object | undefined;
     /**
      * Key of the route which dispatched this action.
      */
diff --git a/package.json b/package.json
index 6346b40ee2cb92899a71685a378e405d5dd00f1c..24422eee11591a891b5b122e964036a6baff557d 100644
--- a/package.json
+++ b/package.json
@@ -21,7 +21,6 @@
   "types": "./lib/typescript/src/index.d.ts",
   "exports": {
     ".": {
-      "source": "./src/index.tsx",
       "types": "./lib/typescript/src/index.d.ts",
       "default": "./lib/module/index.js"
     },
