diff --git a/src/extract/swc/parse.mjs b/src/extract/swc/parse.mjs
index 4015ffeb80fa3af6d7f9444c490c07148b73c8ec..fe0e0523cf0f15a1404f0dc649b37e4c49dbde1b 100644
--- a/src/extract/swc/parse.mjs
+++ b/src/extract/swc/parse.mjs
@@ -17,8 +17,19 @@ const SWC_PARSE_OPTIONS = {
 	target: "es2022",
 	// allow for decorators
 	decorators: true,
-	// TODO: {tj}sx ?
 };
+
+/** @type {ParseOptions} */
+const SWC_TSX_PARSE_OPTIONS = {
+	...SWC_PARSE_OPTIONS,
+	tsx: true,
+};
+
+function getParseOptions(pFileName) {
+	return pFileName.endsWith(".tsx") || pFileName.endsWith(".jsx")
+		? SWC_TSX_PARSE_OPTIONS
+		: SWC_PARSE_OPTIONS;
+}
 /** @type {Map<string, ModuleItem[]>} */
 const CACHE = new Map();
 
@@ -39,7 +50,7 @@ export function getASTCached(pFileName) {
 		return CACHE.get(pFileName);
 	}
 	/** @type {swcCore} */
-	const lAST = swc.parseFileSync(pFileName, SWC_PARSE_OPTIONS);
+	const lAST = swc.parseFileSync(pFileName, getParseOptions(pFileName));
 	CACHE.set(pFileName, lAST);
 	return lAST;
 }
