diff --git a/lib/_tsc.js b/lib/_tsc.js
index 9a9c0ac10f26d2027778f3350d9b62a5fa26ad1f..5bd291b13c8880891855ad7732c5768d6afb3e03 100644
--- a/lib/_tsc.js
+++ b/lib/_tsc.js
@@ -45587,7 +45587,11 @@ function getLocalModuleSpecifier(moduleFileName, info, compilerOptions, host, im
   if (!baseUrl && !paths && !getResolvePackageJsonImports(compilerOptions) || relativePreference === 0 /* Relative */) {
     return pathsOnly ? void 0 : relativePath;
   }
-  const baseDirectory = getNormalizedAbsolutePath(getPathsBasePath(compilerOptions, host) || baseUrl, host.getCurrentDirectory());
+  const pathsBasePath = getPathsBasePath(compilerOptions, host) || baseUrl;
+  if (pathsBasePath === void 0) {
+    return pathsOnly ? void 0 : relativePath;
+  }
+  const baseDirectory = getNormalizedAbsolutePath(pathsBasePath, host.getCurrentDirectory());
   const relativeToBaseUrl = getRelativePathIfInSameVolume(moduleFileName, baseDirectory, getCanonicalFileName);
   if (!relativeToBaseUrl) {
     return pathsOnly ? void 0 : relativePath;
diff --git a/lib/typescript.js b/lib/typescript.js
index 3ba0ce46aad370919b20e29429b3bc72011151a2..b08e8010e4fbf3ecad0bbd08b4bcc0e2ff5934e5 100644
--- a/lib/typescript.js
+++ b/lib/typescript.js
@@ -50207,7 +50207,11 @@ function getLocalModuleSpecifier(moduleFileName, info, compilerOptions, host, im
   if (!baseUrl && !paths && !getResolvePackageJsonImports(compilerOptions) || relativePreference === 0 /* Relative */) {
     return pathsOnly ? void 0 : relativePath;
   }
-  const baseDirectory = getNormalizedAbsolutePath(getPathsBasePath(compilerOptions, host) || baseUrl, host.getCurrentDirectory());
+  const pathsBasePath = getPathsBasePath(compilerOptions, host) || baseUrl;
+  if (pathsBasePath === void 0) {
+    return pathsOnly ? void 0 : relativePath;
+  }
+  const baseDirectory = getNormalizedAbsolutePath(pathsBasePath, host.getCurrentDirectory());
   const relativeToBaseUrl = getRelativePathIfInSameVolume(moduleFileName, baseDirectory, getCanonicalFileName);
   if (!relativeToBaseUrl) {
     return pathsOnly ? void 0 : relativePath;
